src/julia_gc.cc

```diff
@@ -1089,3 +1381,61 @@ void MarkJuliaWeakRef(void * p)
     if (JMarkTyped(p, jl_weakref_type))
         YoungRef++;
 }
+
+
+#ifdef TEST_JULIA_GC_INTERNALS
```

Comment:
When are these tests run? I don't see them being hooked up anywhere.

Reply:
They are run as `__attribute__((constructor)` functions. See the PR description for how to pass the flag. This is a hack in order to be able to run some simple tests.

Reply:
Yeah, but since `TEST_JULIA_GC_INTERNALS` never is `#defined`, that constructor is also never run -- except perhaps on you personal computer, where you added `-DTEST_JULIA_GC_INTERNALS` to your `CPPFLAGS`.

But tests really should be run on Travis CI all the time, to catch future regressions...

Reply:
I totally agree that this is a stopgap measure. I would love to plug them into a testing framework, but as far as I can tell, we only have one for GAP code, not for C code. I could of course register a kernel function for those tests, but that also feels like a workaround.

In short, I'd be delighted to use whatever approach you think is ideal for testing, just let me know what you'd prefer.